polus-precompute-volume-plugin/src/utils.py

```diff
@@ -0,0 +1,338 @@
+            
+            logger.info("\n Iterate through input ...")
+            # Creating a temporary files for the polygon meshes -- will later be converted to Draco
+            with tempfile.TemporaryDirectory() as temp_dir:
+                if not os.path.exists(temp_dir):
+                    os.makedirs(temp_dir, exist_ok=True)
+                for y in range(len(ysplits)-1):
+                    for x in range(len(xsplits)-1):
+                        for z in range(len(zsplits)-1):
+                            start_y, end_y = (ysplits[y], ysplits[y+1])
+                            start_x, end_x = (xsplits[x], xsplits[x+1])
+                            start_z, end_z = (zsplits[z], zsplits[z+1])
+                            
+
+                            volume = bf[start_x:end_x,start_y:end_y,start_z:end_z]
+                            volume = volume.reshape(volume.shape[:3])
+
+                            # Save volume in precomputed format for volumes
+                                # Will build the rest of the pyramid after iterating through input
+                            volume_encoded = ngvol.encode_volume(volume)
+                            highest_res_directory = os.path.join(output_image, str(num_scales))
+                            ngvol.write_image(image=volume_encoded, volume_directory=highest_res_directory, 
+                                            y=(ysplits[y], ysplits[y+1]),
+                                            x=(xsplits[x], xsplits[x+1]),  
+                                            z=(zsplits[z], zsplits[z+1]))
+                            logger.info("Loaded subvolume (YXZ) {}-{}__{}-{}__{}-{}".format(start_y, end_y,
+                                                                                            start_x, end_x,
+                                                                                            start_z, end_z))
+
+                            ids = np.unique(volume)
+                            if (ids == [0]).all():
+                                continue
+                            else:
+                                ids = np.delete(ids, np.where(ids==0))
+                                with ThreadPoolExecutor(max_workers=8) as executor:
+                                    executor.submit(create_plyfiles(subvolume = volume,
+                                                                    ids=ids,
+                                                                    temp_dir=temp_dir,
+                                                                    start_y=start_y,
+                                                                    start_x=start_x,
+                                                                    start_z=start_z,
+                                                                    totalbytes=totalbytes))
+                                    all_identities = np.append(all_identities, ids)
+
+                executor.shutdown(wait=True)
```

Comment:
This is unnecessary. When the code leaves the `ThreadPoolExecutor` block, the executor is closed.